fabric-combat-api-v1/src/main/java/net/fabricmc/fabric/mixin/combat/ProjectileUtilMixin.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.combat;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.ProjectileUtil;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.util.Hand;
+
+import net.fabricmc.fabric.api.combat.v1.bow.FabricBowExtensions;
+
+@Mixin(ProjectileUtil.class)
+public abstract class ProjectileUtilMixin {
+	// Because the uses of this method are hardcoded, checking each hand for the Fabric interfaces of the items is needed.
+	// Note: this does not cancel for the vanilla items unless they are holding a custom implementation of the items
+	@Inject(method = "getHandPossiblyHolding", at = @At(value = "HEAD"), cancellable = true)
+	private static void getHandPossiblyHolding(LivingEntity entity, Item item, CallbackInfoReturnable<Hand> cir) {
+		for (Hand hand : Hand.values()) {
+			if (item == Items.BOW) { // Make sure we only check for bows when searching for bows and allow for other items like crossbows in future
```

Comment:
can we move the `item == Items.BOW` check outside of the for loop? Each time the for loop is called a new array is generated by an Hand.values() call

Reply:
The crossbow also uses this method, and i plan on adding functionality in the same mixin later on

Reply:
Shame that mojang didn't make some VALUES array for hand enum

Reply:
Alright, i made a quick cached constant to not generate it each time the method is called